[Search] Add param type hints (#6490)
sonata-project/SonataAdminBundle: UPGRADE-4.0.md

## RouteCollectionInterface
`RouteCollection` implements `RouteCollectionInterface`.

## SearchHandler

> When no Pager is found, `SearchHandler::search()` now return `null` instead of `false`.

Reviewer: ```suggestion
When no Pager is found, `SearchHandler::search()` returns `null`. Previously, it was returning `false`.
```

Reviewer: IIUC, `SearchHandler::search()` returns `null` when there are no searchable filters, but it doesn't depend on the availability of a pager in any case.

Reviewer 2: Indeed, should be something like this then
```
When there is no searchable filters, `SearchHandler::search()` returns `null`. Previously, it was returning `false`.
```

Can you update the note @tambait ?

Author: Updated
